### DISCUSSION TITLE
how to compile bootloader code?

### DISCUSSION THREAD

**Discussion by HMI1973** [NicoHood/HoodLoader2#127] [Q&A]
I'm new in Arduino Bootloader field and I'm interested in how to compile the bootloader code to generate the HEX file
do you know how could I do this? I tried to google it around and i found that AVRDUDE tools but failed to compile the code with.
Do anyone have idea about how to accomplish this?

Thanks in advance.

**Comment by NicoHood** [comment]
you need avrlibc some devtools and then run make. This only works on linux and also with some avr-gcc versions (sadly). It is very specific, as the bootloader barely fits into the space. I am unable to recompile it with the current avr gcc, as they have changed something unknown in compiling

**Reply by MelvinShwuaner** [reply]
i know this is unrelated, but i cant seem to transfer a string or int between the 16u2 and my main chip, how do you do it?
